### ISSUE TITLE
Pantheon/Terminus upgrade required soon (Sept 2019?)

### ISSUE THREAD

**Issue by rfay** [drud/ddev#1731]
**Describe the bug**

@cyberswat received this from Pantheon; my assumption is that this means we need at least a modest rework of the [go-pantheon](https://github.com/drud/go-pantheon) library @beeradb built: 

> On August 26, 2019, Pantheon will no longer support Terminus versions older than 1.9. We've noticed that in the last month you've used one of these versions, so we wanted to reach out to ask you to update your scripts. Please check all cron jobs or other scripted processes which rely on Terminus to make sure they are on 1.9 or greater.

> We've released Terminus 2.0, hoping to solve many of the concerns of why users haven't upgraded. If your team has questions, concerns, or needs help updating your scripts to the latest Terminus version, please reply to this ticket—we are here to help.

Strategy:
* Try just updating the api version and see if it works
* Postman api study time
* Try to fix the many-sites issue in https://github.com/drud/ddev/issues/1232 while we're there.

**Comment by rfay** [comment]
I think we're going to have to try to get this into v1.10 at the last minute. Since they only gave us a month to do it, that's kind of hard. It could be a small thing, so we'll see how it goes.

**Comment by rfay** [comment]
Had a significant discussion on Pantheon slack with Josh Koenig and Mike Pirog. 

* The Pantheon API remains undocumented
* The Pantheon-recommended technique is to use terminus for access (which is what they originally recommended to us during the initial implementation)
* All we use of the API is auth, site listing, and backup-pulling. Josh says those aren't expected to change immediately
* The 30 day deadline (Aug 26) is when they give themselves permission to break things, it's not when they actually expect to start breaking things. 
* Lando's API client code is at https://github.com/lando/lando/blob/master/plugins/lando-pantheon/lib/client.js
* The cheat technique I tried in https://github.com/drud/go-pantheon/commit/347a0001e756d9b32bde52752a9141b87ef3f3cf is not a valid or useful approach. So I closed #1734
